src/SmartTransactionsController.ts

```diff
@@ -322,23 +322,30 @@ export default class SmartTransactionsController extends StaticIntervalPollingCo
   }
 
   async _executePoll({
-    networkClientId,
+    chainIds,
   }: SmartTransactionsControllerPollingInput): Promise<void> {
     // if this is going to be truly UI driven polling we shouldn't really reach here
     // with a networkClientId that is not supported, but for now I'll add a check in case
     // wondering if we should add some kind of predicate to the polling controller to check whether
     // we should poll or not
-    const chainId = this.#getChainId({ networkClientId });
-    if (!this.#supportedChainIds.includes(chainId)) {
+    // const chainId = this.#getChainId({ networkClientId });
+
+    const filteredChainIds = chainIds.filter((chainId) =>
+      this.#supportedChainIds.includes(chainId),
+    );
+
+    if (filteredChainIds.length === 0) {
       return Promise.resolve();
     }
-    return this.updateSmartTransactions({ networkClientId });
+    return this.updateSmartTransactions({ chainIds: filteredChainIds });
   }
 
   checkPoll({
     smartTransactionsState: { smartTransactions },
   }: SmartTransactionsControllerState) {
-    const currentSmartTransactions = smartTransactions[this.#chainId];
+    const allChainsCurrentTransactions =
+      Object.values(smartTransactions).flat();
+    const currentSmartTransactions = allChainsCurrentTransactions;
```

Comment:
nit: the idea of having the prefix `current` was meant for the currently selected chain. Since this will be for all chains, can we get rid of this line, do this:
```
const smartTransactionsForAllChains = Object.values(smartTransactions).flat();
```
And use `smartTransactionsForAllChains` later on in the code?